nl/tc_linux.go

```diff
@@ -1533,7 +1533,7 @@ func (p *TcPedit) SetIPv6Dst(ip6 net.IP) {
 }
 
 func (p *TcPedit) SetIPv4Src(ip net.IP) {
-	u32 := NativeEndian().Uint32(ip[:4])
+	u32 := NativeEndian().Uint32(ip[12:16])
```

Comment:
Would it not be better to force the input `ip` into v4 notation with `ip.To4()` rather than assuming the passed ip is always in IPv4 in IPv6 notation?

Reply:
good idea! I will fix it.

Reply:
sorry, I only used ipv6 so I didn't check ipv4 before. But thank you for fixing this :)